### PR TITLE
Feature/profile endpoints

### PR DESCRIPTION
**Pull request by RikLitjens**
endpoints + functionality for profile data

### CODE REVIEW

File: src/routes/profile.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        return res.status(200).json(profileData).send('Successfully retrieved profile information');
+    } catch (error) {
+        if (axios.isAxiosError(error)) {
+            if (error.response?.status === 401) {
+                // Unauthorized
+                return res.status(401).send();
+            }
+        }
+        // We cannot handle this specific request, no data for the user.
+        return res.status(503).send();
```
Comment (RinseV) [inline src/routes/profile.ts:30]: We could also send an empty response, but that depends on what the front-end wants/needs

File: src/routes/profile.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+});
+
+/**
+ * Endpoint for posting profile data to gamebus
+ */
+profileRouter.post('/profile', checkJwt, async (req: any, res: any) => {
+    if (!req.query.weight || !req.query.length || !req.query.age) {
```
Comment (RinseV) [inline src/routes/profile.ts:38]: Could also check if ``> 0`` but I digress (+ with JavaScript you never know, a string might be ``> 0``)

File: src/routes/profile.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    // use user info to create a GameBus client
+    const gbClient: GameBusClient = new GameBusClient(
+        new TokenHandler(req.user.accessToken, req.user.refreshToken, req.user.playerId)
+    );
+
+    // extract parameters into bmi model
+    const bmi: BMIModel = {
+        // necessary properties
+        timestamp: new Date().getTime(),
+        weight: parseInt(req.query.weight),
```
Comment (RinseV) [inline src/routes/profile.ts:52]: In GameBus, weight & lenght are doubles, so ``parseFloat`` should be used

File: src/routes/profile.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    // use user info to create a GameBus client
+    const gbClient: GameBusClient = new GameBusClient(
+        new TokenHandler(req.user.accessToken, req.user.refreshToken, req.user.playerId)
+    );
+
+    // extract parameters into bmi model
+    const bmi: BMIModel = {
+        // necessary properties
+        timestamp: new Date().getTime(),
```
Comment (RinseV) [inline src/routes/profile.ts:51]: Check if front-end also wants to send a timestamp, otherwise this works fine

**Review by RinseV — CHANGES_REQUESTED** [review]
Types are not in-line with GameBus spec for POST

File: src/routes/profile.ts (hunk 1 above)
Comment (RikLitjens) [reply src/routes/profile.ts]: Good idea, I've added it for now and asked

File: src/routes/profile.ts (hunk 3 above)
Comment (RikLitjens) [reply src/routes/profile.ts]: Oh that's specific, guess if we want to round it it should be done in frontend

**Review by RinseV — APPROVED** [review]

File: src/routes/profile.ts (hunk 2 above)
Comment (RikLitjens) [reply src/routes/profile.ts]: Sorry, you already approved, but I added this as well

**Pull request merged**